File: src/main/java/me/elsiff/morefish/condition/RainingCondition.java
package me.elsiff.morefish.condition;

import org.bukkit.entity.Player;

public class RainingCondition implements Condition {
    public boolean raining;

    public RainingCondition(boolean raining) {
        this.raining = raining;
    }

    @Override
    public boolean isSatisfying(Player player) {
        return (raining == player.getWorld().hasStorm());
    }
}

File: src/main/kotlin/me/elsiff/morefish/fishing/condition/RainingCondition.kt
package me.elsiff.morefish.fishing.condition

import me.elsiff.morefish.fishing.competition.FishingCompetition
import org.bukkit.entity.Item
import org.bukkit.entity.Player

/**
 * Created by elsiff on 2019-01-12.
 */
class RainingCondition(
    private val raining: Boolean
) : FishCondition {
    override fun check(
        caught: Item,
        fisher: Player,
        competition: FishingCompetition
    ): Boolean {
        return caught.world.hasStorm() == raining
    }
}